src/driver-sinowealth.c

```diff
@@ -358,8 +364,22 @@ sinowealth_init_profile(struct ratbag_device *device)
 static int
 sinowealth_test_hidraw(struct ratbag_device *device)
 {
+	int rc;
+
 	/* Only the keyboard interface has this report */
-	return ratbag_hidraw_has_report(device, SINOWEALTH_REPORT_ID_CONFIG);
+	rc = ratbag_hidraw_has_report(device, SINOWEALTH_REPORT_ID_CONFIG);
+	if (rc)
+		return rc;
+
+	rc = ratbag_hidraw_has_report(device, SINOWEALTH_REPORT_ID_CONFIG_LONG);
+	if (rc) {
+		struct sinowealth_data *drv_data = device->drv_data;
+		drv_data->is_long = true;
+
+		return rc;
+	}
+
+	return 0;
 }
```

Comment:
I'm not sure about this, but I think official software may do such compatibility check in a different way somewhere at the time of launching.
Here's what it does:
1) It checks firmware version (command "05 01"). My G-Wolves Hati returns "V161".
2) Gets the result of this command "05 02" (which returns "05 02 01" for me), true purpose of which I couldn't reverse engineer.
3) Gets the status of lift-off distance and angle snapping ("05 1b"). First bit of the third byte of returned value is always 1, which is suspicious.
4) Gets current configuration, finally using a different report id (6 instead of 4) and getting more data (167 bytes instead of 131).